server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/Mailbox.scala

```diff
@@ -0,0 +1,101 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ * http://www.apache.org/licenses/LICENSE-2.0                   *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ * ***************************************************************/
+
+package org.apache.james.jmap.mail
+
+import org.apache.james.core.Username
+import org.apache.james.jmap.model.UnsignedInt
+import org.apache.james.mailbox.Role
+import org.apache.james.mailbox.model.MailboxId
+
+final case class MailboxName(name: String) {
+  require(!name.isEmpty, "'name' is mandatory")
+}
+
+final case class MailboxRights(mayReadItems: Boolean,
+                               mayAddItems: Boolean,
+                               mayRemoveItems: Boolean,
+                               maySetSeen: Boolean,
+                               maySetKeywords: Boolean,
+                               mayCreateChild: Boolean,
+                               mayRename: Boolean,
+                               mayDelete: Boolean,
+                               maySubmit: Boolean)
+
+object MailboxNamespace {
+  def delegated(owner: Username) = DelegatedNamespace(owner)
+
+  def personal = PersonalNamespace
+}
+
+sealed trait MailboxNamespace {
+  def `type`: String
+  def owner: Option[Username]
+}
+
+case object PersonalNamespace extends MailboxNamespace {
+  override def `type`: String = "Personal"
```

Comment:
I don't see what's the point of this `type` field, the type is already `PersonalNamespace`